cmd/tailscale/cli/set.go

```diff
@@ -76,6 +79,11 @@ func newSetFlagSet(goos string, setArgs *setArgsT) *flag.FlagSet {
 	setf.BoolVar(&setArgs.updateApply, "auto-update", false, "automatically update to the latest available version")
 	setf.BoolVar(&setArgs.postureChecking, "posture-checking", false, "HIDDEN: allow management plane to gather device posture information")
 	setf.BoolVar(&setArgs.runWebClient, "webclient", false, "run a web interface for managing this node, served over Tailscale at port 5252")
+	dap := tailfs.DefaultAutomountPath()
+	if !version.IsSandboxedMacOS() && dap != "" {
```

Comment:
Ideally these flags would still be registered (but maybe not advertised in `--help`) and fail with an helpful error message if used on a different system ("Sorry, $foo isn't available on this system. See URL.") rather than have users read some KB docs, try it on their machine then get an obscure syntax/usage error.

Reply:
I went with something slightly different. I'm leaving the flags available in help but marking them as not supported on this platform.

```
--automount-enabled, --automount-enabled=false
    	(NOT AVAILABLE ON THIS SYSTEM) automatically mount Tailscale shares
  --automount-path string
    	(NOT AVAILABLE ON THIS SYSTEM) path at which to automount shares, leave blank to default to "/Volumes/tailscale"
```

If you attempt to use them on an unsupported system, you get this error message

```
flag automount-enabled is not supported on this system
```

Let me know what you think.